code/modules/atmospherics/machinery/portable/canister.dm

```diff
@@ -415,6 +423,29 @@ GLOBAL_LIST_INIT(gas_id_to_canister, init_gas_id_to_canister())
 		if((10) to (5 * ONE_ATMOSPHERE))
 			. += mutable_appearance(canister_overlay_file, "can-0")
 
+	if(air_pressure != window_pressure_check)
+		window_pressure_check = air_pressure
+		update_window()
+	. += window
```

Comment:
Couldn't this cause false positives? How much of a benefit does it convey?

Reply:
It means that when that canister is idle it wont be constantly updating the window overlay, I'm fine with a bunch of false positives while it's being actively messed with especially since otherwise it would be updating every tick.

Reply:
Canisters call this proc on process_atmos right? There's already a basic level of state control there, tho actually now that I think about it this might break with that. How bad is it without the pressure check?

Reply:
~~I was wrong about it being broken~~

Reply:
Discussed in discord, the pressure check is no longer needed since canisters don't process excessively anymore